grails-app/domain/com/unifina/domain/security/IntegrationKey.groovy

```diff
@@ -50,8 +58,10 @@ class IntegrationKey implements Serializable {
 		if (service == Service.ETHEREUM || service == Service.ETHEREUM_ID) {
 			JSONObject jso = (JSONObject) JSON.parse(json)
 			Map jsmap = [address: jso.get("address")]
-			if(jso.has("privateKeyPlaintext"))
-				jsmap.put("privateKeyPlaintext", jso.get("privateKeyPlaintext"))
+			if(service == Service.ETHEREUM) {
+				String decryptedPrivateKey = getEthereumIntegrationKeyService().decryptPrivateKey(this)
```

Comment:
To me, feels like caching the the `ethereumIntegrationKeyService` is an unnecessary micro-optimization, probably not worth the slight code pollution. I don't see a problem with:

`String decryptedPrivateKey = Holders.getApplicationContext().getBean(EthereumIntegrationKeyService.class).decryptPrivateKey(this)`